spec/models/activity_pub/actor_spec.cr

```diff
@@ -295,26 +297,32 @@ Spectator.describe ActivityPub::Actor do
 
     it "adds a public following relationship" do
       foo_bar.follow(other, confirmed: true, visible: true).save
-      expect(foo_bar.all_following(public: true)).to eq([other])
-      expect(foo_bar.all_following(public: false)).to eq([other])
+      expect(foo_bar.all_following(public: true).size).to eq(1)
+      expect(foo_bar.all_following(public: true).first.iri).to eq(other.iri)
+      expect(foo_bar.all_following(public: false).size).to eq(1)
+      expect(foo_bar.all_following(public: false).first.iri).to eq(other.iri)
```

Comment:
A bunch of these tests were testing exact matches on the entire object. This was failing because the DB has a default value of JSON `[]` for attachments, so once it was saved it was `attachments = []` rather than `attachments = nil`. Converted to check that the result has 1 item and that the `iri`s match.

Reply:
if the column is defined to accept null (see my longer reply to the PR) then these changes won't be necessary.

Reply:
Yeah, I really wanted to not have to accept null, but as I recall, the framework wouldn't compile with that defined as `not null`, and then there is the whole issue where you can't set a default value in the DB and have it respected. I will look at backing out that change, although since I have it running this way on my site it will involve some gymnastics to back it out. A default value of empty array is much better than null, but it seems to be swimming upstream.

Reply:
removing it shouldn't break your site and existing records. your code already handles `nil` in the right places, and loading an empty array from existing records should just work—it looked like you already reduced the case of four empty slots in the array to an empty array (if not it would be easy to handle) so that one edge case should be okay, as well.

i'm not opposed to the default being empty array, btw. i'm just trying to keep the surface area of changes small so that i can more easily reason about them.

Reply:
Sorry, just meant that I have to manually copy the table data over to a new table and rename back, because sqlite doesn't support live schema modifications other than additive operations.